sdk/nodejs/utils.ts

```diff
@@ -94,31 +94,12 @@ export function promiseResult<T>(promise: Promise<T>): T {
 }
 
 /**
- * Lifts the properties of the value a promise resolves to and adds them to the promise itself. This
- * can be used to take a function that was previously async (i.e. Promise-returning) and make it
- * synchronous in a backward compatible fashion.  Specifically, because the function now returns a
- * `Promise<T> & T` all properties on it will be available for sync consumers, while async consumers
- * can still use `await` on it or call `.then(...)` on it.
+ * No longer supported. This function is now a no-op and will directly return the promise passed
+ * into it.
  *
  * This is an advanced compat function for libraries and should not generally be used by normal
  * Pulumi application.
  */
 export function liftProperties<T>(promise: Promise<T>, opts: InvokeOptions = {}): Promise<T> & T {

```

Comment:
This helps ensure that we don't have to update our existing data sources.  For context, the existing data sources call:

```ts
    const promise: Promise<GetRestApiResult> = pulumi.runtime.invoke("aws:apigateway/getRestApi:getRestApi", {
        "name": args.name,
    }, opts);

    return pulumi.utils.liftProperties(promise, opts);
```

Because `pulumi.runtime.invoke` will actually return synchronously and will *already* have lifted the values such that you can get to them both as properties (and through the Promise shape), there is no need to `liftProperties` anymore.  So `liftProperties` is now a no-op.